docs/advanced-topics/serialization/inetworkserializable.md

```diff
@@ -73,32 +72,114 @@ The below `INetworkSerializable` implementation example works only with RPCs and
 [Read More About Custom NetworkVariable Implementations](../../basics/networkvariable.md)
 :::
 
+Because of the limitations of NetworkVariable, which is implemented to avoid creating garbage as much as possible, C# arrays aren't supported in NetworkVariable, including arrays that are embedded in structs, as they prevent those structs from being allocated without creating garbage. Unity provides alternatives to C# arrays, however, that do work with NetworkVariable - `NativeArray<>` and `NativeList<>`.
+
+`NativeArray<>` works similarly to C# arrays in that they are statically sized, with the size determined at allocation, while `NativeList<>` behaves similarly to `List<>` in being dynamically resizable. `NativeList<>` requires importing the `collections` package, but it does have one advantage over `NativeArray<>` in that it has a return-by-ref `ElementAt()` method. Since the `BufferSerializer` API is bi-directional serialization that takes values by ref for both reading and writing, this method simplifies serialization of array values and avoids the need to copy to and from temporary values.
+
+Care must be taken when using `NativeArray<>` and `NativeList<>`, however, because they do need to be manually allocated and deallocated. When using them, you must ensure that they've been properly initialized before use, and that they're disposed when no longer used. Take note, as well, that copying a `NativeArray<>` or `NativeList<>` isn't a true copy - like C# classes, both copies will reference the same data, so if you make copies of your structs, make sure that they are only disposed once.
```

Comment:
Let's put this in some kind of `Info` type box and make it a bit more generic. Something like "Be aware of object lifetime, especially for types implementing IDisposable (like NativeArray)."

I'm also not sure if this belongs in this doc or the NetworkVariable doc? Maybe a short notice here and then a longer explanation goes into NetworkVariable.